superset/models/core.py

```diff
@@ -801,7 +801,8 @@ def get_quoter(self):
 
     def get_df(self, sql, schema):
         sqls = [str(s).strip().strip(';') for s in sqlparse.parse(sql)]
-        engine = self.get_sqla_engine(schema=schema)
+        engine = self.get_sqla_engine(
+            schema=schema, source=utils.sources.get('chart', None))
```

Comment:
So sources.get('sql_lab') will already return a `None` you don't have to explicitly set it here